packages/dev/gui/src/2D/controls/container.ts

```diff
@@ -619,6 +621,42 @@ export class Container extends Control {
         this._measureForChildren.copyFrom(this._currentMeasure);
     }
 
+    public isWidthFullyDefined(): boolean {
+        if (!this._isDirty && this._cachedIsWidthFullyDefined !== undefined) {

```

Comment:
If we update a child control, will the `_isDirty` flag of the container control be updated?

When `this.adaptWidthToChildren` is `true`, I think we will have to re-run the loop below to recompute the value of `_cachedIsWidthFullyDefined`, so `_isDirty` should be set to `true` in this case.

Reply:
Good point! I'll check first if any of the children have isDirty set to true before early returning

Reply:
Yes, I don't see a performant way to do it....

If you have to loop over all the children to check the `_isDirty` flag, you might as well remove that flag/cache and check `isDimensionFullyDefined` directly, I don't think that will be much slower than testing the `_isDirty` flag.

So, instead of using `_isDirty` and managing a cache, perhaps you should add a flag to enable or disable this test? It would be `true` by default to be backward compatible, but we would indicate in the documentation that to gain performance, the flag should be set to `false`. Or, to avoid loosing performance, we set the flag to `false` by default. Even if it breaks backward compatibility, it's only for a warning message that would not show by default...

Let's see what the others think.

Reply:
_isDirty is used in _layout well. I wouldn't change that without fully testing it. Setting it to true and not changing it will trigger layout changes on each frame (unless you make changes to the layout).
I guess we could make it a getter that iterate the tree and returns true if any of the children is dirty? There is no way around it, and there is a slight performance hit for sure.

Reply:
I would not want to risk breaking things for a better warning... So, if no other solutions, I would vote for the "flag" solution as described in my answer above (and I would set it `false` by default).

Reply:
I think the flag idea would be best, and we don't even need to create a new flag, because `ignoreLayoutWarnings` should already do this job, right?

Reply:
All the better if we already have a flag for that!